src/xb-silo-query.c

```diff
@@ -190,23 +232,29 @@ xb_silo_query_part (XbSilo *self,
 		    GPtrArray *results,
 		    GHashTable *results_hash,
 		    XbQuery *query,
+		    XbQueryContext *context,
+		    gboolean first_result_only,
```

Comment:
could this be in XbSiloQueryHelperFlags?

Reply:
Not easily, that would mean exposing `XbSiloQueryHelperFlags` in `xb-silo-query-private.h` and shifting a lot of stuff around.

At this point I’d really like to just get these changes merged, and then further cleanups like this can be done as-needed! The changes in this commit are all in private APIs, so there’s no problem with API breaks if we do changes there in future.

Reply:
Point taken.